pyproject.toml

```
@@ -1,12 +1,173 @@
+[tool.cibuildwheel]
+build = [
+    "cp37-*",
+    "cp38-*",
+    "cp39-*",
+    "cp310-*",
+]
+
+    [tool.cibuildwheel.environment]
+    PIP_PREFER_BINARY = true
+
+    [tool.cibuildwheel.windows]
+    archs = [
+        "AMD64",
+    ]
+
+
+    [tool.cibuildwheel.macos]
+    archs = [
+        "x86_64",
+
+        # Unfortunately we don't have Apple Silicon support yet
+        # TODO: Fix this
+        # Current problem seems to be that installed libfftw3 does not provide arm64
+        # symbols
+        # "universal2",
+        # "arm64",
+    ]
+
+        [tool.cibuildwheel.macos.environment]
+        CC = "/usr/bin/clang"
+        CXX = "/usr/bin/clang++"
+        CFLAGS = "-Wno-implicit-function-declaration -I/usr/local/opt/fftw/include"
+        LDFLAGS = "-Wl,-rpath,/usr/local/opt/fftw/lib -L/usr/local/opt/fftw/lib"
```

Not sure if it is worth the effort for PyFFTW as there is a lot less Cython code than in scikit-image, but it is possible to shrink the size of the binary wheels by stripping debug symbols. For clan this would be adding `-Wl,-S ` here.

```suggestion
        LDFLAGS = "-Wl,-S -Wl,-rpath,/usr/local/opt/fftw/lib -L/usr/local/opt/fftw/lib"
```
At least for scikit-image this shrinks wheel size by more than a factor of two.

I ended up doing that for Linux as well via gcc's ``-Wl,--strip-debug`, but I struggled a bit getting the environment variable to be found. I didn't get it to work with LDFLAGS, but it worked when defining a [custom environment variable](https://github.com/scikit-image/scikit-image/blob/1a14bb32b5a6718870dc0b3df186b7ed3018f337/.github/workflows/wheel_tests_and_release.yml#L51-L52) that was [checked in setup.py](https://github.com/scikit-image/scikit-image/blob/1a14bb32b5a6718870dc0b3df186b7ed3018f337/setup.py#L119-L120).

---

Just checked the wheels and they are all around 1.5-3 MB so this is probably not too high of a priority. For scikit-image we had sizes >30MB and we able to shrink to around 13MB.

---

Is this meant to be `-Wl,-S -Wl` and not `-S -Wl`? Would those not be the same?

---

No, the `-Wl,-S` must be specified together as suggested here. Options to be passed on from the compiler to the linker are prefixed with "-Wl,". There is a lot more info [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/developer_guide/gcc-using-libraries) under the section "Passing the -Wl option" (That is for gcc but I think it is the same for clang). From that description it could also be concatenated into a single `-Wl` with 3 options afterward:  `-Wl,-S,-rpath,/usr/local/opt/fftw/lib`.

On second look, I would have thought it would have an `=` instead of `,` between `-rpath` and `/usr/local/opt/fftw/lib`, but [answers here state that either form is fine](https://stackoverflow.com/questions/6562403/i-dont-understand-wl-rpath-wl).

---

Ah, that makes sense - thank you!

I changed it to `"-Wl,-S,-rpath=/usr/local/opt/fftw/lib -L/usr/local/opt/fftw/lib"` which I think is the most reading-friendly version.

Leaving this open due to:

> I ended up doing that for Linux as well via gcc's ``-Wl,--strip-debug`, but I struggled a bit getting the environment variable to be found. I didn't get it to work with LDFLAGS, but it worked when defining a custom environment variable that was checked in setup.py.

---

@grlee77 interestingly I only see a reduction of ≈ 10 KB out of the ≈ 3 MB wheel on MacOS.

I will leave in the MacOS flags, but won't bother with Linux since the reduction is negligible.